types/node/index.d.ts

```diff
@@ -748,6 +748,11 @@ declare module "http" {
     export interface ServerRequest extends IncomingMessage {
         connection: net.Socket;
     }
+
+    export interface ServerResponseHeaders {
+      [key: string]: number | string | string[];
```

Comment:
Numbers are allowed as input, see the type of the Content-Length header in the example; https://nodejs.org/api/http.html#http_response_writehead_statuscode_statusmessage_headers

Everything that can be converted to a primitive value is technically valid input.